zilliqa/src/exec.rs

```diff
@@ -678,9 +685,43 @@ impl State {
             .collect())
     }
 
+    pub fn get_stakers_bytes(&self) -> Result<Vec<Vec<u8>>> {
+        let data = contracts::deposit::GET_STAKERS.encode_input(&[])?;
+
+        let stakers = self.call_contract(
+            Address::ZERO,
+            Some(contract_addr::DEPOSIT),
+            data,
+            0,
+            // The chain ID and current block are not accessed when the native balance is read, so we just pass in some
+            // dummy values.
+            0,
+            BlockHeader::default(),
+        )?;
+
+        let stakers = contracts::deposit::GET_STAKERS
+            .decode_output(&stakers)
+            .unwrap()[0]
+            .clone()
+            .into_array()
+            .unwrap();
```

Comment:
All of this seems duplicated with `get_stakers()`, and I think worth extracting into a helper function.

In fact, no, `get_stakers()` already calls `into_bytes()` inside its map, so I'd probably even have `get_stakers()` just call `get_stakers_bytes()` and re-iter the result constructing `NodePublicKey`s from the vec. This iterates one more time but I think that shouldn't be an issue since all performance-critical paths will probably use the `_bytes()` version anyway - and it should make the code much more concise.